packages/react-components/react-spinbutton/.swcrc

```diff
@@ -0,0 +1,39 @@
+{
```

Comment:
I compared the built output from this branch to master because I was curious to see where the bundle size savings were coming from. I only compared SpinButton since I'm familiar with the code but I expect these findings hold true across the repo.

1. Built output is generally identical though there are some changes as the new build doesn't transform things like `??` (which is a win!)
2. In `renderSpinButton` object spread is transformed to Object.assign() in the new build.

### SWC `renderSpinButton` output

SWC also doesn't line break this like the previous build which isn't really an issue but that's what I put a single line here.

```js
return /*#__PURE__*/React.createElement(slots.root, Object.assign({}, slotProps.root), /*#__PURE__*/React.createElement(slots.input, Object.assign({}, slotProps.input)), /*#__PURE__*/React.createElement(slots.incrementButton, Object.assign({}, slotProps.incrementButton)), /*#__PURE__*/React.createElement(slots.decrementButton, Object.assign({}, slotProps.decrementButton)));
```

### `master` branch `renderSpinButton` output

```js
return /*#__PURE__*/React.createElement(slots.root, {
    ...slotProps.root
  }, /*#__PURE__*/React.createElement(slots.input, {
    ...slotProps.input
  }), /*#__PURE__*/React.createElement(slots.incrementButton, {
    ...slotProps.incrementButton
  }), /*#__PURE__*/React.createElement(slots.decrementButton, {
    ...slotProps.decrementButton
  }));
```

According to caniuse [object spread is within our support matrix](https://caniuse.com/mdn-javascript_operators_spread_spread_in_object_literals) so transforming it seems like the wrong thing.

I wrote up a [simple benchmark to compare spread to Object.assign()](https://jsbench.me/zrlf1d88kx/1) and spread is much faster in Chromium while the two approachs are roughly the same in Firefox and Safari (tested on iOS)

Reply:
For reference, this is basically the helper function that's being used 

```
    function extends_() {
        return extends_ = Object.assign || function(target) {
            for (var i = 1; i < arguments.length; i++) {
                var source = arguments[i];
                for (var key in source) Object.prototype.hasOwnProperty.call(source, key) && (target[key] = source[key]);
            }
            return target;
        }, extends_.apply(this, arguments);
    }

```

And this is how swc uses it:

```
    /*#__PURE__*/
    return external_React_namespaceObject.createElement(slots.root, function() {
        return extends_.apply(this, arguments);
    }({}, slotProps.root));
```

Reply:
Good to know! I filed an [issue with SWC](https://github.com/swc-project/swc/issues/7044) since it seems to be a bug as spread only appears to be transformed in the context of React.